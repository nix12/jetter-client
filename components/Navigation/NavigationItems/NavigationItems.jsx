import React from 'react';

import Menu from '@material-ui/core/Menu';
import NavigationItem from './NavigationItem/NavigationItem';

const navigationItems = props => {
  const { name, onClose } = props;
  const [anchorEl, setAnchorEl] = React.useState(null);

  return (
    <Menu {...props}>
      <NavigationItem
        click={onClose}
        link="/user/show"
        linkAs={`/u/${name}`}
        userName={name}
      >
        My Profile
      </NavigationItem>

      <NavigationItem
        click={onClose}
        link="/user/update"
        linkAs={`/u/${name}/password`}
      >
        Change Password
      </NavigationItem>

      <NavigationItem click={onClose} link="/auth/logout" linkAs="/logout">
        Logout
      </NavigationItem>
    </Menu>
  );
};

export default navigationItems;
