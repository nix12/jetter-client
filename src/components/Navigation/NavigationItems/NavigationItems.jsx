import React from 'react';
import Cookies from 'universal-cookie';
import jwtDecode from 'jwt-decode';

import Menu from '@material-ui/core/Menu';
import NavigationItem from './NavigationItem/NavigationItem';

const navigationItems = props => {
  const { name, onClose } = props;
  const cookies = new Cookies();
  const token = cookies.get('token');
  const data = token ? jwtDecode(token) : '';
  const { username } = data;
  const [anchorEl, setAnchorEl] = React.useState(null);

  return (
    <Menu {...props}>
      <NavigationItem
        click={onClose}
        link="/user/[username]"
        linkAs={`/user/${username}`}
        userName={username}
      >
        My Profile
      </NavigationItem>

      <NavigationItem
        click={onClose}
        link="/user/update"
        linkAs={`/u/${name}/password`}
      >
        Change Password
      </NavigationItem>

      <NavigationItem click={onClose} link="/logout" linkAs="/logout">
        Logout
      </NavigationItem>
    </Menu>
  );
};

export default navigationItems;
