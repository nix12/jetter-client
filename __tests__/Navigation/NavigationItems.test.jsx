import React from 'react';
import { shallow } from 'enzyme';
import NavigationItems from '../../components/Navigation/NavigationItems/NavigationItems';

describe('navigationItem', () => {
  it('should render navigationItems component', () => {
    expect(shallow(<NavigationItems />)).toMatchSnapshot();
  });
});
