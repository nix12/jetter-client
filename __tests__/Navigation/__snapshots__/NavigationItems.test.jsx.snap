// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navigationItem should render navigationItems component 1`] = `ShallowWrapper {}`;
